lib/Doctrine/Common/Annotations/DocParser.php

```diff
@@ -859,7 +865,36 @@ private function Annotation()
             );
         }
 
-        $values = $this->MethodCall();
+        $arguments = $this->MethodCall();
+
+        $positionalArguments = $arguments['positional_arguments'] ?? [];
+        $namedArguments      = $arguments['named_arguments'] ?? [];
+
+        if (
+            self::$annotationMetadata[$name]['has_named_argument_constructor']
+            && self::$annotationMetadata[$name]['default_property'] !== null
+        ) {
+            $values = $namedArguments;
+            foreach (self::$annotationMetadata[$name]['constructor_args'] as $property => $parameter) {
+                $position = $parameter['position'];
+                if (isset($values[$property]) || ! isset($positionalArguments[$position])) {
+                    continue;
+                }
+
+                $values[$property] = $positionalArguments[$position];
+            }
+        } else {
+            $values = $namedArguments;
+            if (! empty($positionalArguments) && ! isset($values['value'])) {
```

Comment:
Please use count instead, empty does not communicate intent well

Reply:
Isn't `count() > 0` much less effective than `!empty()` ?

Reply:
Yes, but the output of this code is usually cached and even if not, this is not a tight loop

Reply:
I wanted to toy with phpbench so I build a little benchmark from [this raw benchmark](https://stackoverflow.com/a/29635982/353612):

```php
<?php

class ArrayCheckerBench
{
    public function provideArrays(): \Generator
    {
        yield 'empty array' => ['array' => []];
        yield 'huge array' => ['array' => range(0, 100000)];
    }

    public function checkArrayWithEmpty(array $x): bool
    {
        return empty($x);
    }

    public function checkArrayWithCount(array $x): bool
    {
        return count($x) === 0;
    }

    /**
     * @Revs(1000000)
     * @Iterations(5)
     * @ParamProviders({"provideArrays"})
     */
    public function benchCheckIfArrayIsEmptyWithEmpty(array $parameters): void
    {
        $this->checkArrayWithEmpty($parameters['array']);
    }

    /**
     * @Revs(1000000)
     * @Iterations(5)
     * @ParamProviders({"provideArrays"})
     */
    public function benchCheckIfArrayIsEmptyWithCount(array $parameters): void
    {
        $this->checkArrayWithCount($parameters['array']);
    }
}
```

Here are the results:

```
vendor/bin/phpbench run ArrayCheckerBench.php --report=aggregate --retry-threshold=0.5
PhpBench @git_tag@. Running benchmarks.

\ArrayCheckerBench

    benchCheckIfArrayIsEmptyWithEmpty # emp.I4 [μ Mo]/r: 0.285 0.285 (μs) [μSD μRSD]/r: 0.000μs 0.06%
    benchCheckIfArrayIsEmptyWithEmpty # hug.R1 I1 [μ Mo]/r: 0.287 0.287 (μs) [μSD μRSD]/r: 0.000μs 0.12%
    benchCheckIfArrayIsEmptyWithCount # emp.R1 I1 [μ Mo]/r: 0.309 0.309 (μs) [μSD μRSD]/r: 0.001μs 0.28%
    benchCheckIfArrayIsEmptyWithCount # hug.R3 I3 [μ Mo]/r: 0.311 0.311 (μs) [μSD μRSD]/r: 0.001μs 0.34%

2 subjects, 20 iterations, 4,000,000 revs, 0 rejects, 0 failures, 0 warnings
(best [mean mode] worst) = 0.285 [0.298 0.298] 0.285 (μs)
⅀T: 5.958μs μSD/r 0.001μs μRSD/r: 0.201%
suite: 134621f1b80d1bb3f7ed3eaf6bc32c7bb639d7d1, date: 2021-02-07, stime: 15:08:37
+-------------------+-----------------------------------+-------------+---------+-----+-------------+---------+---------+---------+---------+---------+--------+-------+
| benchmark         | subject                           | set         | revs    | its | mem_peak    | best    | mean    | mode    | worst   | stdev   | rstdev | diff  |
+-------------------+-----------------------------------+-------------+---------+-----+-------------+---------+---------+---------+---------+---------+--------+-------+
| ArrayCheckerBench | benchCheckIfArrayIsEmptyWithEmpty | empty array | 1000000 | 5   | 526,184b    | 0.285μs | 0.285μs | 0.285μs | 0.285μs | 0.000μs | 0.06%  | 1.00x |
| ArrayCheckerBench | benchCheckIfArrayIsEmptyWithEmpty | huge array  | 1000000 | 5   | 64,194,552b | 0.287μs | 0.287μs | 0.287μs | 0.288μs | 0.000μs | 0.12%  | 1.01x |
| ArrayCheckerBench | benchCheckIfArrayIsEmptyWithCount | empty array | 1000000 | 5   | 526,184b    | 0.308μs | 0.309μs | 0.309μs | 0.310μs | 0.001μs | 0.28%  | 1.08x |
| ArrayCheckerBench | benchCheckIfArrayIsEmptyWithCount | huge array  | 1000000 | 5   | 64,194,552b | 0.310μs | 0.311μs | 0.311μs | 0.313μs | 0.001μs | 0.34%  | 1.09x |
+-------------------+-----------------------------------+-------------+---------+-----+-------------+---------+---------+---------+---------+---------+--------+-------+
```

Looks like count takes almost 10% more time, so in relative terms, it's far better indeed. In absolute terms though, it seems to be 30 ns slower… I don't think we should worry about such small differences.